File: app/src/main/java/io/github/erikcaffrey/kata_dagger2_mariokart/view/ui/PrincessPeachActivity.java
package io.github.erikcaffrey.kata_dagger2_mariokart.view.ui;

import android.support.annotation.DrawableRes;
import android.support.annotation.StringRes;
import android.support.v4.content.ContextCompat;
import android.widget.ImageView;
import butterknife.BindView;
import io.github.erikcaffrey.kata_dagger2_mariokart.PrincessPeachApplication;
import io.github.erikcaffrey.kata_dagger2_mariokart.R;
import io.github.erikcaffrey.kata_dagger2_mariokart.data.PrincessPeachApi;
import io.github.erikcaffrey.kata_dagger2_mariokart.domain.PrincessPeach;
import io.github.erikcaffrey.kata_dagger2_mariokart.view.widget.MarioKartLabel;
import javax.inject.Inject;

public class PrincessPeachActivity extends BaseActivity {

  @Inject PrincessPeachApi princessPeachApi;

  @BindView(R.id.label_peach) MarioKartLabel peachLabel;
  @BindView(R.id.picture_peach) ImageView peachImage;

  @Override protected int getLayoutResID() {
    return R.layout.activity_princess_peach;
  }

  @Override protected void onPrepareActivity() {
    super.onPrepareActivity();
    initializeDagger();
    PrincessPeach princessPeach = getPrincesPeachFromApi();
    renderPrincesPeach(princessPeach);
  }

  private void renderPrincesPeach(PrincessPeach princessPeach) {
    renderName(princessPeach.getName());
    renderPicture(princessPeach.getPhoto());
  }

  private void renderName(@StringRes int name) {
    peachLabel.setText(name);
  }

  private void renderPicture(@DrawableRes int picture) {
    peachImage.setImageDrawable(ContextCompat.getDrawable(this, picture));
  }

  private PrincessPeach getPrincesPeachFromApi() {
    return princessPeachApi.getPrincesPeach();
  }

  private void initializeDagger() {
    PrincessPeachApplication application = (PrincessPeachApplication) getApplication();
    application.getPrincessPeachComponent().inject(this);
  }
}

File: app/src/main/java/io/github/erikcaffrey/kata_dagger2_mariokart/view/ui/PrincessPeachActivity.kt
package io.github.erikcaffrey.kata_dagger2_mariokart.view.ui

import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import io.github.erikcaffrey.kata_dagger2_mariokart.PrincessPeachApplication
import io.github.erikcaffrey.kata_dagger2_mariokart.R
import io.github.erikcaffrey.kata_dagger2_mariokart.data.PrincessPeachApi
import io.github.erikcaffrey.kata_dagger2_mariokart.domain.PrincessPeach
import kotlinx.android.synthetic.main.activity_princess_peach.*
import javax.inject.Inject

class PrincessPeachActivity : AppCompatActivity() {

  @Inject lateinit var princessPeachApi: PrincessPeachApi

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_princess_peach)
    initializeDagger()
  }

  override fun onResume() {
    super.onResume()
    renderPrincesPeach(princessPeachApi.princesPeach)
  }

  private fun renderPrincesPeach(princessPeach: PrincessPeach) {
    label_peach.setText(princessPeach.name)
    picture_peach.setImageDrawable(ContextCompat.getDrawable(this, princessPeach.photo))
  }

  private fun initializeDagger() {
    val application = application as PrincessPeachApplication
    application.princessPeachComponent.inject(this)
  }
}
